packages/react-dom/src/__tests__/ReactDOMFloat-test.js

```diff
@@ -552,6 +552,53 @@ describe('ReactDOMFloat', () => {
     });
   });
 
+  describe('document encapsulation', () => {
+    // @gate enableFloat
+    it('can support styles inside shadowRoots', async () => {
+      const shadow = document.body.attachShadow({mode: 'open'});
```

Comment:
A more interesting test (in addition) might be to use a child within the shadow root to show that it's not just that the container might be a shadow root but it might also be inside a shadow root.

Reply:
Pushed a new test that I think covers what you are asking about here